sympy/solvers/tests/test_ode.py

```diff
@@ -3574,6 +3573,13 @@ def test_factorable():
     assert set(sols) == set(dsolve(eq, f(x), hint='factorable'))
     assert checkodesol(eq, sols) == 2*[(True, 0)]
 
+    eq = (f(x).diff(x)**2-1)*(f(x).diff(x)**2-4)
+    sols = [Eq(f(x), C1 - x), Eq(f(x), C1 + x), Eq(f(x), C1 + 2*x), Eq(f(x), C1 - 2*x)]
+    assert set(sols) == set(dsolve(eq, f(x), hint='factorable'))
+    assert checkodesol(eq, sols) == 4*[(True, 0)]
+
+    eq = (f(x).diff(x, 2)-exp(f(x)))*(f(x).diff(x, 2)+exp(f(x)))
+    raises(NotImplementedError, lambda: dsolve(eq, hint = 'factorable'))
```

Comment:
You've changed this from what I suggested which was `dsolve((f(x).diff(x, 2)-exp(f(x)))*f(x).diff(x))` The point was supposed to be that we can solve one factor but no the other. What does this PR do in that situation?

Reply:
If any of the factors is not solvable then only it will raise `NotImplementedError`.

Reply:
I guess you mean "all of the factors" rather than "any".

If one is solvable and the other is not do we get a solution from the one that is solvable? There should be a test for that.

Reply:
> I guess you mean "all of the factors" rather than "any".


 yes

Reply:
> If one is solvable and the other is not do we get a solution from the one that is solvable? There should be a test for that.
okay

Reply:
> > If one is solvable and the other is not do we get a solution from the one that is solvable? There should be a test for that.
> >

 okay